example/lib/onegini_listener.dart

```diff
@@ -161,22 +147,66 @@ class OneginiListener extends OneginiEventListener {
   }
 
   @override
-  void openCustomTwoStepRegistrationScreen(
+  void eventInitCustomRegistration(
+    BuildContext buildContext, String data) {
+
+    try {
+      var response = jsonDecode(data);
+      var providerId = response["providerId"];
+
+      if (providerId == "2-way-otp-api") {
+        // a 2-way-otp does not require data for the initialization request
+        OneginiCustomRegistrationCallback()
+          .submitSuccessAction(providerId, null)
+          .catchError((error) => {
+            if (error is PlatformException) {
+              showErrorToast(error.message)
+            }
+          });
+      }
+    } on FormatException catch (error) {
+      showErrorToast(error.message);
+      return;
+    }
+  }
+
+  @override
+  void eventFinishCustomRegistration(
       BuildContext buildContext, String data) {
-    var response = jsonDecode(data);
-    if (response["providerId"] == "2-way-otp-api")
-      Navigator.push(
-        buildContext,
-        MaterialPageRoute(
-            builder: (context) => OtpScreen(
-                  password: response["data"],
-                )),
-      );
+
+    try {
+      var response = jsonDecode(data);
+      var providerId = response["providerId"];
+
+      if (providerId == "2-way-otp-api")
```

Comment:
Reused strings across the class, like `2-way-otp-api` can be extracted to variable.

Reply:
This is within the example app, meant as an example; the hardcoded "2-way-otp-api" works better as a code snippet example to demonstrate that there should be a string. Because of this, I prefer the strings as is. (open for discussion but this is what i am thinking atm)

Reply:
Ok, so you think people will copy-paste the code and it's gonna be easier if they do not care about the variable declared somewhere at the top of the file. Makes sense 👍